osu.Game/Screens/Select/FooterV2/FooterButtonV2.cs

```diff
@@ -0,0 +1,194 @@
+
+        private Colour4 buttonAccentColour;
+
+        protected Colour4 AccentColour
+        {
+            set
+            {
+                buttonAccentColour = value;
+                bar.Colour = buttonAccentColour;
+                icon.Colour = buttonAccentColour;
+            }
+        }
+
+        protected IconUsage Icon
+        {
+            set => icon.Icon = value;
+        }
+
+        protected LocalisableString Text
+        {
+            set => text.Text = value;
+        }
+
+        private readonly SpriteText text;
+        private readonly SpriteIcon icon;
+
+        protected Container TextContainer;
+        private readonly Box bar;
+        private readonly Box backgroundBox;
+
+        public FooterButtonV2()
+        {
+            EdgeEffect = new EdgeEffectParameters
+            {
+                Type = EdgeEffectType.Shadow,
+                Radius = 5,
+                Roundness = 10,
+                Colour = Colour4.Black.Opacity(0.25f)
+            };
+            Shear = SHEAR;
+            Size = new Vector2(button_width, button_height);
+            Masking = true;
+            CornerRadius = corner_radius;
+            Children = new Drawable[]
+            {
+                backgroundBox = new Box
+                {
+                    RelativeSizeAxes = Axes.Both
+                },
+
+                //For elements that should not be sheared.
+                new Container
+                {
+                    Anchor = Anchor.CentreLeft,
+                    Origin = Anchor.CentreLeft,
+                    Shear = -SHEAR,
+                    RelativeSizeAxes = Axes.Both,
+                    Children = new Drawable[]
+                    {
+                        TextContainer = new Container
+                        {
+                            Anchor = Anchor.TopCentre,
+                            Origin = Anchor.TopCentre,
+                            Y = 42,
+                            AutoSizeAxes = Axes.Both,
+                            Child = text = new OsuSpriteText
+                            {
+                                //figma design says the size is 16, but due to the issues with font sizes 19 matches better
+                                Font = OsuFont.TorusAlternate.With(size: 19),
+                                AlwaysPresent = true
+                            }
+                        },
+                        icon = new SpriteIcon
+                        {
+                            Y = 12,
+                            Size = new Vector2(20),
+                            Anchor = Anchor.TopCentre,
+                            Origin = Anchor.TopCentre
+                        },
+                    }
+                },
+                new Container
+                {
+                    // The X offset has to multiplied as such to account for the fact that we only want to offset by the distance from the CenterLeft point of the container
```

Comment:
Which X offset? There's none specified.

Reply:
Oversight. Referenced some jank positioning I had before . Will remove in a bit when I can get to my pc